File: MealPlanService/MealPlanService.Application/DTOs/Updating/UpdateMealPlanDTO.cs
using MealPlanService.Domain.Entities;
using MealPlanService.Domain.Enums;

namespace MealPlanService.Application.DTOs
{ 
    public class UpdateMealPlanDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public MealPlanType Type { get; set; }

        public List<MealPlanDayDTO> Days { get; set; } = new List<MealPlanDayDTO>();
    }
}

File: NutritionBalance.ClientFront/src/app/models/mealPlan/DTOs/update-meal-plan.dto.ts
import { MealPlanDayDTO } from './meal-plan-day.dto';
import { MealPlanType } from '../enums/meal-plan-type.enum';

export class UpdateMealPlanDTO {
  id!: number;
  name!: string;
  description!: string;
  type!: MealPlanType;
  days: MealPlanDayDTO[] = [];
}
